### DISCUSSION TITLE
what should i put in the google sheets reference key ? I have tried everything but it seems not working

### DISCUSSION THREAD

**Discussion by zac202002** [AutomaApp/automa#316] [Q&A]
what should i put in the google sheets reference key ? I have tried everything but it seems not working
Can someone  have an example for me ? Q_Q

**Comment by Delusionnnn** [comment]
Sorry this is old, but maybe you still need help?

Whatever you want to use to reference the data. 

For example I have "data" as the reference key and use it as the pull request for the Loop Data block. Then use {{loopData.loopID.path}} to use the looping data in a form block.

By other standards you can get the raw data with {{googleSheets.referencekey.1}} with the number being what block index number you want the data from.

**Reply by AI-Based-Businesses** [reply]
I'm trying your {{googleSheets.referencekey.1}} format but it isn't working for me, but {{googleSheets.referencekey}} works and it gives me all the data from the Google Sheets block, but I need data only at a single index...

**Comment by Delusionnnn** [comment]
> I'm trying your {{googleSheets.referencekey.1}} format but it isn't working for me, but {{googleSheets.referencekey}} works and it gives me all the data from the Google Sheets block, but I need data only at a single index...

Ah I see what you mean. Ya it is similar to an issue I had. You can use {{googleSheets.referencekey.1}} will work as long as you uncheck the "first row as keys" check mark in the google sheets and then find your reference number (make sure your range is set correctly otherwise it'll be hard to find. Trouble with this method is that it'll pull the raw javascript so XQYZ will paste as ["XQYZ"] and I can't figure out how to stop that. 

**However there is a work around** and will work for your case. Use the loop data block. Assign the loop data block the same reference key as the google sheets block you want to reference. Then use {{loopData.loopID.path}} and set the max data to loop to 1 and start from index you will have to find by running the script and finding the loop data index number in the logs. (For example, if a column with the header being a reference key the first block down will be index 0, then next block is index 1, and so on. 

You can see how I am referencing it below in my workflow as an example in a forms block. Lmk if you have any questions but as long as you have loop data set to 1 you shouldn't need a loop break block but its a good idea to have one just in case it messes up the code.

![image](https://github.com/AutomaApp/automa/assets/141815049/1ee62fc1-5870-47ff-92e0-8d6223293d11)
![image](https://github.com/AutomaApp/automa/assets/141815049/f119781c-8ec6-43f2-ad44-ffe75125ab4b)
![image](https://github.com/AutomaApp/automa/assets/141815049/36d67eae-508c-4dba-a905-fcf45b450d1e)

**Reply by AI-Based-Businesses** [reply]
Thanks for the detailed reply! It's apparent that there's no way to make the index format work for Google Sheets block references, even though it's mentioned in the docs, except for the case of using the sheet headers as a sort of indices, as is shown in Automa's YouTube video "Autorial: Part 4." Instead, I just used a Loop Data block and then referenced that with index format, similar to what you've done but with just numbers instead of headers, and, although it's more complicated, it actually works.

**Reply by AI-Based-Businesses** [reply]
Since we seem to disagree that the index notation works at all, maybe it's that it works for columns but not rows, as I'm pulling a row into my workflow and the index notation doesn't work at all, let alone with JavaScript.

**Reply by Delusionnnn** [reply]
I have not tried rows so could be the reason yes. It would probably work if you just selected the range to be that one block but it'll still pull the javascript notation so its basically useless and slice variable wont remove it. 

Using Loop Data seems to work as of now as a workaround, so I'm glad you were able to get it resolved :) 
(trust me I know the pain it took me like 7 hours to figure all the looping stuff out)
